contrib/wireshark/ucx_le64.lua

```diff
@@ -0,0 +1,1683 @@
+        if connect_to_ep ~= 0 then
+            tree:add_le(utf.cm_req_map_key, cm_id)
+        else
+            tree:add_le(utf.cm_req_conn_sn, cm_id)
+        end
+        tree:add(utf.cm_req_sockaddr, tvbuf)
+    end
+
+    if connect_to_ep ~= nil then
+        msg = msg .. ", Con-To-EP: " .. connect_to_ep
+        if connect_to_ep == 0 then
+            msg = msg .. ", Map-Key: " .. cm_id:le_int64()
+        else
+            msg = msg .. ", Seq: " .. cm_id:le_int64()
+        end
+    end
+    if tvbuf ~= nil then
+        orig_tree:append_text(", Sockaddr: " .. tvbuf:bytes():tohex())
+    end
+    return msg
+end
+
+function uct_am_dissector_stream(tvbuf, pinfo, tree)
+    ucp_stream_proto.dissector(tvbuf:tvb(), pinfo, tree)
+end
+
+function uct_tcp_am_dissector_put_req(tvbuf, pinfo, tree)
+    local addr = tvbuf(0, 8)
+    local len = tvbuf(8, 8)
+    local sn = tvbuf(16, 4)
+
+    local tree = tree:add("Put Req")
+    tree:add_le(uct_tcp_fields.put_req_addr, addr)
+    tree:add_le(uct_tcp_fields.put_req_len, len)
+    tree:add_le(uct_tcp_fields.put_req_sn, sn)
+
+    local length = len:le_uint64():tonumber()
+    if length > 0 then
+        local data = tvbuf(20, length)
+        tree:add_le(uct_tcp_fields.put_data, data)
+    end
+
+    local addr_str = string.format("0x%s", addr:le_uint64():tohex())
+    local msg = "Seq: " .. sn:le_uint()
+            .. ", Mem Addr: " .. addr_str .. ", Data Len: " .. len:le_uint64()
+
+    return msg
+end
+
+function uct_tcp_am_dissector_put_ack(tvbuf, pinfo, tree)
+    local sn = tvbuf(0, 4)
+
+    local tree = tree:add("Put Ack")
+    tree:add_le(uct_tcp_fields.put_ack_sn, sn)
+
+    local msg = "Seq: " .. sn:le_uint()
+    return msg
+end
+
+function uct_tcp_am_dissector_atomic_req(tvbuf, pinfo, tree)
+    ucp_amo_sw_proto.dissector(tvbuf:tvb(), pinfo, tree)
+end
+
+function uct_tcp_am_dissector_atomic_rep(tvbuf, pinfo, tree)
+    pinfo.private.atomic_rep = 1
+    ucp_amo_sw_proto.dissector(tvbuf:tvb(), pinfo, tree)
+end
+
+function uct_tcp_am_compl(tvbuf, pinfo, tree)
+    ucp_amo_sw_proto_cmpl_dissector(tvbuf:tvb(), pinfo, tree)
+end
+
+function uct_am_rndv_rtr(tvbuf, pinfo, tree)
+    ucp_rndv_proto_dissect_rtr(tvbuf:tvb(), pinfo, tree)
+end
+
+function uct_am_rndv_rts(tvbuf, pinfo, tree)
+    ucp_rndv_proto_dissect_rts(tvbuf:tvb(), pinfo, tree)
+end
+
+function uct_am_rndv_atp(tvbuf, pinfo, tree)
+    ucp_rndv_proto_dissect_atp(tvbuf:tvb(), pinfo, tree)
+end
+
+function uct_am_rndv_ats(tvbuf, pinfo, tree)
+    ucp_rndv_proto_dissect_ats(tvbuf:tvb(), pinfo, tree)
+end
+
+local uct_am_dissector = {
+    [1]  = uct_tcp_am_dissector_wireup,
+
+    [9]  = uct_am_rndv_rts,
+    [10] = uct_am_rndv_ats,
+    [11] = uct_am_rndv_rtr,
+
+    [15] = uct_am_dissector_stream,
+    [16] = uct_am_rndv_atp,
+
+    -- Atomic SW, on top of AM
+    [20] = uct_tcp_am_dissector_atomic_req,
+    [21] = uct_tcp_am_dissector_atomic_rep,
+    [22] = uct_tcp_am_compl,
```

Comment:
it also looks agnostic to a UCT transport - no need for tcp in the name

Reply:
Agree, removed `tcp` everywhere.